staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -240,14 +267,43 @@ func (s *sharedIndexInformer) GetController() Controller {
 	return &dummyController{informer: s}
 }
 
-func (s *sharedIndexInformer) AddEventHandler(handler ResourceEventHandler) error {
+func (s *sharedIndexInformer) AddEventHandler(handler ResourceEventHandler) {
+	s.AddEventHandlerWithResyncPeriod(handler, s.defaultEventHandlerResyncPeriod)
+}
+
+func determineResyncPeriod(desired, check time.Duration) time.Duration {
+	if desired == 0 {
+		return desired
+	}
+	if check == 0 {
+		glog.Warningf("The specified resyncPeriod %v is invalid because this shared informer doesn't support resyncing", desired)
+		return 0
+	}
+	if desired < check {
+		glog.Warningf("The specified resyncPeriod %v is being increased to the minimum resyncCheckPeriod %v", desired, check)
+		return check
+	}
+	if desired%check != 0 {
+		// e.g. if resyncPeriod is 5s and resyncCheckPeriod is 3s, we want (floor(5/3) * 3) + 3 = 6
```

Comment:
This doesn't match my expectation.  If I want 5 and it checks every three, seems like I would just check if my 5 seconds was done each time.  Why try to make them match as a multiple?  It will reduce smoothness, but it seems like its closer to correct.